_posts/2024-09-26-managing-connector-offsets.md

```diff
@@ -0,0 +1,271 @@
+When using a `KafkaMirrorMaker2` resource, the configurations for the ConfigMaps for listing and altering offsets, as well as the `state` configuration for stopping and resuming the connector, are specified on a per-connector basis. For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker-2
+spec:
+  # ...
+  mirrors:
+  - # ...
+    sourceConnector:
+      listOffsets:
+        toConfigMap:
+          name: my-connector-offsets
+      alterOffsets:
+        fromConfigMap:
+          name: my-connector-offsets
+      state: stopped
+    # ...
+```
+
+Strimzi allows you to list, alter and reset offsets for all the MirrorMaker connectors (`MirrorSourceConnector`, `MirrorCheckpointConnector`, `MirrorHeartbeatConnector`).
+However, be aware that the only connector that currently actively uses its offsets is the `MirrorSourceConnector`.
+Listing offsets can be useful for the `MirrorCheckpointConnector` and `MirrorHeartbeatConnector` to track progress.
+However, altering or resetting offsets for these connectors is rarely necessary.
+
+Strimzi only allows you to perform actions on one connector within a single mirror at a time.
+Therefore, to initiate an action from a `KafkaMirrorMaker2` resource you must apply two annotations:
+* `strimzi.io/connector-offsets`
+* `strimzi.io/mirrormaker-connector`
+
+Set `strimzi.io/connector-offsets` to one of `list`, `alter` or `reset`.
+At the same time set `strimzi.io/mirrormaker-connector` to the name of your connector.
+
+Strimzi names the connectors using the format `<SOURCE_ALIAS>-><TARGET_ALIAS>.<CONNECTOR_TYPE>`, for example `east-kafka->west-kafka.MirrorSourceConnector`.
+
+You can use a single command to annotate the resource with both annotations.
+For example, this command lists offsets for a connector called `east-kafka->west-kafka.MirrorSourceConnector`:
+
+```shell
+$ kubectl annotate kafkaconnector my-source-connector strimzi.io/connector-offsets=list strimzi.io/mirrormaker-connector="east-kafka->west-kafka.MirrorSourceConnector" -n kafka
+```
+
+When listing and altering offsets for MirrorMaker connectors, Strimzi uses the connector name in the data field.
+For example, the above command to list offsets for the connector `east-kafka->west-kafka.MirrorSourceConnector` results in a ConfigMap containing:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+# ...
+data:
+  east-kafka--west-kafka.MirrorSourceConnector.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            "cluster": "east-kafka",
+            "partition": 0,
+            "topic": "mirrormaker2-cluster-configs"
+          },
+          "offset": {
+            "offset": 0
+          }
+        }
+      ]
+    }
+```
+
+### Conclusion
+
+Now you can list, alter, and reset offsets of your connectors using the `KafkaConnector` and `KafkaMirrorMaker2` custom resources.
```

Comment:
That's a very short conclusion. Maybe we can say the use cases mentioned in the intro (if you follow my other suggestion above) are now easily done?

Reply:
Does the new conclusion work better?